pkg/controllers/framework.go

```diff
@@ -72,11 +75,22 @@ func (km *KindControllerManager) add(source string, ev api.EventType, fns []Cont
 func (km *KindControllerManager) Handle(id string) {
 
 	ctx := context.Background()
+	logger := logger.NewOCMLogger(ctx)
 
-	// TODO: lock the Event with a fail-fast advisory lock context.
+	// lock the Event with a fail-fast advisory lock context.
 	// this allows concurrent processing of many events by one or many controller managers.
 	// allow the lock to be released by the handler goroutine and allow this function to continue.
 	// subsequent events will be locked by their own distinct IDs.
+	lockOwnerID, acquired, err := km.lockFactory.NewUnblockingLock(ctx, id, db.Events)
+	if err != nil {
+		logger.Error(fmt.Sprintf("Error obtaining the event lock: %v", err))
+		return
+	}
+	if !acquired {
+		logger.Infof("Event %s is processed by another worker, continue to process the next", id)
+		return
+	}
+	defer km.lockFactory.Unlock(ctx, lockOwnerID)
 	threadContext := context.WithValue(ctx, "event", id)
 
 	// for now ... single threaded execution. need locks.
```

Comment:
this comment can be removed.

Reply:
parallel execution happens by having more than 1 replica of our deployed pod.

Reply:
removed